types/src/transaction/metadata.rs

```diff
@@ -1,8 +1,14 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+//! This file implements Libra transaction metadata types to allow
```

Comment:
@sblackshear @n4ss Is the concept of "metadata" specific to the peer-to-peer script, or meant to be shared between several Move scripts?

In the former case, we may want to choose names that leave space for the next move script with a different kind of metadata, say concretely, replacing "Metadata" with wither "TransferMetadata" or "PeerToPeerMetadata".

Reply:
Afaik, metadata might be relevant for more use cases but only PeerToPeer for now